DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/ManageCourseController.cs

```diff
@@ -99,5 +114,98 @@ EditLearningPathwayDefaultsViewModel model
 
             return RedirectToAction("Index", new { customisationId = model.CustomisationId });
         }
+
+        [HttpGet]
+        [ServiceFilter(typeof(RedirectEmptySessionData<EditLearningPathwayDefaultsData>))]
+        [Route("AutoRefreshOptions")]
+        public IActionResult EditAutoRefreshOptions(int customisationId)
+        {
+            var centreId = User.GetCentreId();
+            var categoryId = User.GetAdminCategoryId()!;
+
+            var courseDetails = courseService.GetCourseDetailsForAdminCategoryId(
+                customisationId,
+                centreId,
+                categoryId.Value
+            );
+
+            var model = new EditAutoRefreshOptionsViewModel(courseDetails!);
+
+            SetViewBagCourseOptions(customisationId, model.RefreshToCustomisationId);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ServiceFilter(typeof(RedirectEmptySessionData<EditLearningPathwayDefaultsData>))]
+        [Route("AutoRefreshOptions")]
+        public IActionResult SaveAutoRefreshOptions(
+            int customisationId,
+            EditAutoRefreshOptionsViewModel model
+        )
+        {
+            if (customisationId != model.CustomisationId)
+            {
+                return new StatusCodeResult(500);

```

Comment:
I don't think the customisationId should even be posted with the model. It would be a part of the ViewModel not the FormData (see earlier comment about ViewBag for documentation).

Reply:
I've split the FormData from the View Model now :)